github/copilot.go

```diff
@@ -0,0 +1,263 @@
+}
+
+// SeatAssignments represents the number of seats assigned.
+type SeatAssignments struct {
+	SeatsCreated int64 `json:"seats_created"`
+}
+
+// SeatCancellations represents the number of seats cancelled.
+type SeatCancellations struct {
+	SeatsCancelled int64 `json:"seats_cancelled"`
+}
+
+func (cp *CopilotSeatDetails) UnmarshalJSON(data []byte) error {
+	type Alias CopilotSeatDetails
+	var seatDetail Alias
+
+	if err := json.Unmarshal(data, &seatDetail); err != nil {
+		return err
+	}
+
+	cp.AssigningTeam = seatDetail.AssigningTeam
+	cp.PendingCancellationDate = seatDetail.PendingCancellationDate
+	cp.LastActivityAt = seatDetail.LastActivityAt
+	cp.LastActivityEditor = seatDetail.LastActivityEditor
+	cp.CreatedAt = seatDetail.CreatedAt
+	cp.UpdatedAt = seatDetail.UpdatedAt
+
+	switch v := seatDetail.Assignee.(type) {
+	case map[string]interface{}:
+		jsonData, err := json.Marshal(seatDetail.Assignee)
+		if err != nil {
+			return err
+		}
+		if v["type"].(string) == "User" {
+			user := &User{}
+			if err := json.Unmarshal(jsonData, user); err != nil {
+				return err
+			}
+			cp.Assignee = user
+		} else if v["type"].(string) == "Team" {
+			team := &Team{}
+			if err := json.Unmarshal(jsonData, team); err != nil {
+				return err
+			}
+			cp.Assignee = team
+		} else if v["type"].(string) == "Organization" {
+			organization := &Organization{}
+			if err := json.Unmarshal(jsonData, organization); err != nil {
+				return err
+			}
+			cp.Assignee = organization
+		} else {
+			return fmt.Errorf("unsupported assignee type %s", v["type"].(string))
+		}
+	default:
+		return fmt.Errorf("unsupported assignee type %T", v)
+	}
+
+	return nil
+}
+
+// GetCopilotBilling Gets Copilot for Business seat information and settings for an organization
+//
+// GitHub API docs: https://docs.github.com/en/rest/copilot/copilot-for-business#get-copilot-for-business-seat-information-and-settings-for-an-organization
+func (s *CopilotService) GetCopilotBilling(ctx context.Context, org string) (*OrganizationCopilotDetails, *Response, error) {
+	u := fmt.Sprintf("orgs/%v/copilot/billing", org)
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var copilotDetails *OrganizationCopilotDetails
+	resp, err := s.client.Do(ctx, req, &copilotDetails)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return copilotDetails, resp, nil
+}
+
+// ListCopilotSeats Gets Copilot for Business seat assignments for an organization
+//
+// GitHub API docs: https://docs.github.com/en/rest/copilot/copilot-for-business#list-all-copilot-for-business-seat-assignments-for-an-organization
+func (s *CopilotService) ListCopilotSeats(ctx context.Context, org string) (*CopilotSeats, *Response, error) {
+	u := fmt.Sprintf("orgs/%v/copilot/billing/seats", org)
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var copilotSeats *CopilotSeats
+	resp, err := s.client.Do(ctx, req, &copilotSeats)
```

Comment:
How are we handling pagination here for more than 100 users?

Reply:
@o-sama I feel happy to help you on the paginations one and other stuff here, because I am developing the same for my own stuff.

Reply:
pagination works in current version